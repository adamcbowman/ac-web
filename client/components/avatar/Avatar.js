import React, { PureComponent } from 'react'
import PropTypes from 'prop-types'
import classnames from 'classnames/bind'
import styles from './Avatar.css'
import { initials } from 'utils/string'
import { Loading } from 'react-powerplug'

export default class Avatar extends PureComponent {
    static PropTypes = {
        name: PropTypes.string.isRequired,
        url: PropTypes.string,
        size: PropTypes.number,
    }
    static defaultProps = {
        size: 60,
    }
    constructor(props) {
        super(props)

        this.classnames = classnames.bind(styles)
    }
    get style() {
        const { size } = this.props

        return {
            height: size,
            width: size,
            fontSize: size < 50 ? '0.75em' : '1em',
        }
    }
    renderer = ({ isLoading, setLoading }) => {
        const { url, name } = this.props
        const classNames = this.classnames({
            Initials: isLoading,
            Avatar: !isLoading,
        })

        return (
            <div
                className={classNames}
                data-initials={initials(name)}
                style={this.style}>
                {url && (
                    <img
                        src={url}
                        alt={initials(name)}
                        title={name}
                        onLoad={() => setLoading(false)}
                        onError={() => setLoading(false)}
                    />
                )}
            </div>
        )
    }
    render() {
        const loading = Boolean(this.props.url)

        return <Loading initial={loading}>{this.renderer}</Loading>
    }
}
