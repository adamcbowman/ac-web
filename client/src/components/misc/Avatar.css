@import "../../styles/colors.css";

.Avatar {
    width: 40px;
    height: 40px;
    overflow: hidden;
    margin: 0 auto;
    background-color: var(--brand-primary);
    border-radius: 20px;
    color: white;

    & > img {
        height: 100%;
        width: 100%;
    }

    &:after {
        content: attr(data-initials);
        color: white;
        font-size: 1rem;
        line-height: 2.5rem;
        display: inline-block;
        text-align: center;
        width: 100%;
        vertical-align: middle;
    }
}


.Initials {
    composes: Avatar;
    & > img {
        display: none;
    }
}
