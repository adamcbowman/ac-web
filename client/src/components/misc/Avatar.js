import React, {PropTypes} from 'react'
import { compose, setDisplayName } from 'recompose'
import CSSModules from 'react-css-modules'
import styles from './Avatar.css'
import loadingState from './loadingState'
import {getInitials} from 'utils/string'

Avatar.propTypes = {
    name: PropTypes.string.isRequired,
    url: PropTypes.string,
}

function Avatar({ url, name, onLoad, onError, isLoaded, hasError, isLoading }) {
    const initials = getInitials(name)
    const styleName = (hasError === true || isLoading === true) ? 'Initials' : 'Avatar'

    return (
        <div styleName={styleName} data-initials={initials}>
            <img src={url} alt={initials} title={name} onLoad={onLoad} onError={onError} />
        </div>
    )
}

export default compose(
    setDisplayName('Avatar'),
    loadingState,
)(CSSModules(Avatar, styles))
