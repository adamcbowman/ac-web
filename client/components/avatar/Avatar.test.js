import React from 'react'
import Renderer from 'react-test-renderer'
import Avatar from './Avatar'

test('avatar component', () => {
    let avatar = Renderer.create(<Avatar name="Karl Guillotte" />)

    expect(avatar).toMatchSnapshot()

    avatar = Renderer.create(
        <Avatar
            name="Karl Guillotte"
            url="//avatars1.githubusercontent.com/u/744011?v=3&s=40"
        />
    )

    expect(avatar).toMatchSnapshot()
})
