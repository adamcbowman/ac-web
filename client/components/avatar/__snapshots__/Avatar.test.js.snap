// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`avatar component 1`] = `
<div
  className="Avatar"
  data-initials="KG"
  style={
    Object {
      "fontSize": "1em",
      "height": 60,
      "width": 60,
    }
  }
/>
`;

exports[`avatar component 2`] = `
<div
  className="Initials"
  data-initials="KG"
  style={
    Object {
      "fontSize": "1em",
      "height": 60,
      "width": 60,
    }
  }
>
  <img
    alt="KG"
    onError={[Function]}
    onLoad={[Function]}
    src="//avatars1.githubusercontent.com/u/744011?v=3&s=40"
    title="Karl Guillotte"
  />
</div>
`;
